src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,213 @@
+    val n_c = DenseVector[Double](n_current._1, n_current._2, n_current._3)
+    val n_d = DenseVector[Double](n_desired._1, n_desired._2, n_desired._3)
+    if (math.abs(linalg.norm(linalg.cross(n_c,n_d))) > NumericUtils.EPSILON) {
+      val v = linalg.cross(n_c, n_d)
+      val s = linalg.norm(v)
+      val c = n_c dot n_d
+
+      val v_skew = DenseMatrix.zeros[Double](3,3)
+      v_skew(0,1) = -v(2)
+      v_skew(0,2) = v(1)
+      v_skew(1,0) = v(2)
+      v_skew(1,2) = -v(0)
+      v_skew(2,0) = -v(1)
+      v_skew(2,1) = v(0)
+
+      DenseMatrix.eye[Double](3) + v_skew + (v_skew * v_skew) * (1-c)/(s*s)
+    } else {
+      DenseMatrix.eye[Double](3)
+    }
+  }
+
+  /**
+    * Finds the center of a list of vertices - defined as the average coordinate of all the vertices in the list
+    * @param vertices List of vertices
+    * @return Coordinate of center of vertices
+    */
+  private def findCenter(vertices: Seq[(Double, Double, Double)]): (Double, Double, Double) = {
+    (vertices.map(_._1).sum / vertices.length.toDouble,
+      vertices.map(_._2).sum / vertices.length.toDouble,
+      vertices.map(_._3).sum / vertices.length.toDouble)
+  }
+
+  /**
+    * Compares pointA to pointB. If pointA is first relative to pointB rotating counter-clockwise about center,
+    * the method returns true. Input vectors should have same z-coordinate - comparison is based in 2-D
+    * @param pointA Point of interest
+    * @param pointB Point of comparison
+    * @param center Reference point for comparison
+    * @return Returns TRUE if pointA is first relative to pointB, FALSE otherwise
+    */
+  private def ccwCompare(pointA: (Double, Double, Double), pointB: (Double, Double, Double),
+                         center: (Double, Double, Double)): Boolean = {
+    // Check that points are in the same x-y plane
+    assert(math.abs(pointA._3 - pointB._3) < NumericUtils.EPSILON)
+
+    // Starts counter-clockwise comparison from 12 o'clock. Center serves as origin and 12 o'clock is along
+    // vertical line running through this center.
+    // Check if points are on opposite sides of the center. Points left of center will be before points
+    // right of the center
+    if ((pointA._1 - center._1 < -NumericUtils.EPSILON) && (pointB._1 - center._1 >= NumericUtils.EPSILON)) {
+      return true
+    }
+    else if ((pointA._1 - center._1 >= NumericUtils.EPSILON) && (pointB._1 - center._1 < -NumericUtils.EPSILON)) {
+      return false
+    }
+
+    // Compares points that fall on the x = center._1 line.
+    if ((math.abs(pointA._1 - center._1) < NumericUtils.EPSILON) &&
+        (math.abs(pointB._1 - center._1) < NumericUtils.EPSILON)) {
+      // Points furthest away from the center will be before points that are closer to the center
+      if ((pointA._2 - center._2 >= NumericUtils.EPSILON) || (pointB._2 - center._2 >= NumericUtils.EPSILON)) {
+        return pointA._2 > pointB._2
+      }
+      return pointB._2 > pointA._2
```

Comment:
could you put this inside an `else` just to make it more obvious what's going on?